src/client/push-client.js

```diff
@@ -175,6 +174,9 @@ export default class PushClient extends EventDispatch {
    */
   unsubscribe() {
     return this.getRegistration()
+    .catch(() => {
```

Comment:
I think I'd move this down to just before the final `.then`. If the catch happens here then we know that the next two `.then`s are NOOPs, so might as well not bother running them.